### PR TITLE
New LP on PTQ/QAT in ExecuTorch

### PR DESCRIPTION
**Pull request by annietllnd**
Wait for author feedback before merging

### CODE REVIEW

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+title: Understanding PTQ and QAT
+weight: 2
+
+### FIXED, DO NOT MODIFY
+layout: learningpathall
+---
+
+## PTQ vs QAT: what changes in practice?
+
+PTQ and QAT both aim to run your model with quantized operators (typically INT8). The difference is where you pay the cost: PTQ optimizes for speed of iteration, while QAT optimizes for quality and robustness.
```
Comment (Burton2000) [inline content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md:11]: We should probably have the terms written out in full to start with before using the acronyms everywhere.

E.g.
Post-training quantization (PTQ) and Quantization-aware training (QAT)

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+---
+title: Understanding PTQ and QAT
```
Comment (Burton2000) [inline content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md]: Thoughts on some more detail in the title, e.g.

Understanding quantization with ExecuTorch and the Arm backend

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+layout: learningpathall
+---
+
+## PTQ vs QAT: what changes in practice?
+
+PTQ and QAT both aim to run your model with quantized operators (typically INT8). The difference is where you pay the cost: PTQ optimizes for speed of iteration, while QAT optimizes for quality and robustness.
+
+In this Learning Path, you use quantization as part of the ExecuTorch Arm backend. The goal is to export a quantized model that can run on Arm hardware with dedicated neural accelerators (NX).
+
+To keep the workflow concrete, you start with a complete, runnable CIFAR-10-based example that exports `.vgf` artifacts end to end. After you have a known-good baseline, you can apply the same steps to your own upscaler model and training loop.
```
Comment (Burton2000) [inline content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md]: to your own neural network and training code.

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+In a nutshell, the Arm backend in ExecuTorch consists of the following building blocks:
+
+- TOSA (Tensor Operator Set Architecture) provides a standardized operator set for acceleration on Arm platforms.
```
Comment (Burton2000) [inline content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md:19]: provides an open, standardized, minimal operator set for neural networks operations to be lowered to. It is utilized by Arm platforms and accelerators.

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+- The ExecuTorch Arm backend lowers your PyTorch model to TOSA and uses an ahead-of-time (AOT) compilation flow.
+- The VGF backend produces a portable artifact you can carry into downstream tools, including `.vgf` files.
+
+### Post-training quantization (PTQ)
+
+PTQ keeps training simple. You train your FP32 model as usual, then run a calibration pass using representative inputs to determine quantization parameters (for example, scales). After calibration, you convert the model and export a quantized graph.
+
+PTQ is a good default when you need a fast iteration loop and you have a calibration set that looks like the actual inference data. For upscalers, PTQ can be good enough for early bring-up, especially when your goal is to validate the export and integration path.
```
Comment (Burton2000) [inline content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md]: For neural networks

Comment (Burton2000) [reply content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md]: I would also add that PTQ, depending on the model and use-case can still provide good quality results equal to the original floating point graph.

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+### Quantization-aware training (QAT)
+
+QAT simulates quantization effects during training. You prepare the model for QAT, fine-tune with fake-quantization enabled, then convert and export.
+
+QAT is worth the extra effort when PTQ introduces visible artifacts. This is common for image-to-image tasks because small numeric changes can show up as banding, ringing, or loss of fine detail.
```
Comment (Burton2000) [inline content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md]: introduces visible drop in model accuracy. For example, this is common for

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+## How this maps to the Arm backend
+
+For Arm-based platforms, the workflow stays consistent across models:
+
+1. Train and evaluate the upscaler in PyTorch.
```
Comment (Burton2000) [inline content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md]: 1. Train and evaluate the neural network in PyTorch.

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+2. Quantize (PTQ or QAT) to reduce runtime cost.
+3. Export through TOSA and generate a `.vgf` artifact.
```
Comment (Burton2000) [inline content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md]: 3. Export with ExecuTorch (via TOSA) to generate a `.vgf` artifact.

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/3-run-ptq-and-export-vgf.md [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+
+## Overview
+
+In this section, you apply post-training quantization (PTQ) to an image-to-image model and export a `.vgf` artifact.
+
+The default workflow in this Learning Path is end to end: you run a complete CIFAR-10-based example, generate a `.vgf` artifact, and validate that the Arm backend export path works on your machine.
+
+After that, you take the same PTQ export logic and apply it to your own model and calibration data.
+
+## Run the end-to-end PTQ example (CIFAR-10)
```
Comment (Burton2000) [inline content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/3-run-ptq-and-export-vgf.md]: I would just remove the CIFAR-10 part as this is just the dataset and not the task

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/4-run-qat-and-export-vgf.md [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+
+{{% notice Tip %}}
+If export fails with a missing model converter error, you likely forgot to source the Arm backend `setup_path.sh` in your current terminal session.
+{{% /notice %}}
+
+As the script runs, you should see QAT training logs (prefixed with `qat`). When export completes, you should see `.vgf` output under `./output_qat/`.
+
+## Advanced: drop-in QAT export to VGF for your own project
+
+If PTQ introduces visible artifacts, QAT is the next step. The workflow is the same as PTQ, but you insert a short fine-tuning phase after you prepare the model for QAT.
```
Comment (Burton2000) [inline content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/4-run-qat-and-export-vgf.md]: If PTQ degrades model accuracy too much,

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/5-validate-and-choose-a-quantization-strategy.md [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+The fastest way to understand the integration constraints is to start from a known-good sample and then replace the model.
+
+Use the Learning Path [Get started with neural graphics using ML Extensions for Vulkan](/learning-paths/mobile-graphics-and-gaming/vulkan-ml-sample/) and focus on how the sample loads and executes `.vgf` artifacts. This is where you validate assumptions about input and output tensor formats and where any required color-space or layout conversions happen.
+
+## Wrap-up
+
+You now have a complete reference workflow for quantizing an image-to-image model with TorchAO and exporting INT8 `.vgf` artifacts using the ExecuTorch Arm backend. You also have a practical baseline you can use to debug export issues before you switch to your production model and data.
+
+When you move from the CIFAR-10 proxy model to your own upscaler, keep these constraints in mind:
```
Comment (Burton2000) [inline content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/5-validate-and-choose-a-quantization-strategy.md]: +When you move from the CIFAR-10 proxy model to your own model, keep these constraints in mind:

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md (hunk 1 above)
Comment (annietllnd) [reply content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md:11]: This is done on the starting page for the LP which will be displayed before the introduction, but good catch! In _index.md if you're curious

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md (hunk 2 above)
Comment (annietllnd) [reply content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md]: Great suggestion. Changed the main title to Overview and then the first subsection to your suggestion, to avoid truncation on smaller screens

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md (hunk 3 above)
Comment (annietllnd) [reply content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md]: Done

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md (hunk 4 above)
Comment (annietllnd) [reply content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md:19]: Done

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md (hunk 5 above)
Comment (annietllnd) [reply content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md]: Done

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md (hunk 6 above)
Comment (annietllnd) [reply content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md]: Done

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md (hunk 7 above)
Comment (annietllnd) [reply content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md]: Done

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md (hunk 8 above)
Comment (annietllnd) [reply content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md]: Done

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/3-run-ptq-and-export-vgf.md (hunk 9 above)
Comment (annietllnd) [reply content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/3-run-ptq-and-export-vgf.md]: Good point, done

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/4-run-qat-and-export-vgf.md (hunk 10 above)
Comment (annietllnd) [reply content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/4-run-qat-and-export-vgf.md]: Done

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/5-validate-and-choose-a-quantization-strategy.md (hunk 11 above)
Comment (annietllnd) [reply content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/5-validate-and-choose-a-quantization-strategy.md]: Done

File: content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md (hunk 1 above)
Comment (Burton2000) [reply content/learning-paths/mobile-graphics-and-gaming/quantize-neural-upscaling-models/1-introduction.md:11]: ah nice thanks for confirming!

**Review by Burton2000 — APPROVED** [review]
LGTM
